mongosql/src/algebrizer/definitions.rs

```diff
@@ -475,14 +477,61 @@ impl<'a> Algebrizer<'a> {
         }
 
         // Build the Project Stage using the source and built expression.
-        let stage = mir::Stage::Project(mir::Project {
+        let project_stage = mir::Stage::Project(mir::Project {
             source: Box::new(source),
-            expression,
+            expression: expression.clone(),
             is_add_fields,
             cache: SchemaCache::new(),
         });
-        stage.schema(&self.schema_inference_state())?;
-        Ok(stage)
+        project_stage.schema(&self.schema_inference_state())?;
+        Ok(if is_distinct {
+            let group_stage = Self::build_distinct_values_group_stage(
+                &expression,
+                project_stage,
+                self.scope_level,
+            );
+            group_stage.schema(&self.schema_inference_state())?;
+            group_stage
```

Comment:
Aren't we supposed to follow-up the `$group` stage with a `$project` that removes the `_id` and projects the original datasources?

Reply:
I added the `$project` to get the original datasources.